src/Seabug/Contract/Util.purs

```diff
@@ -78,6 +80,20 @@ modify fn t = wrap (fn (unwrap t))
 
 data ReturnBehaviour = ToMarketPlace | ToCaller
 
+payBehaviour
+  :: ReturnBehaviour
+  -> ValidatorHash
+  -> (Value.CurrencySymbol /\ Value.TokenName)
+  -> TxConstraints Void Void
+payBehaviour ToCaller _ _ = mempty -- Balancing will return the token to the caller
+payBehaviour ToMarketPlace valHash asset =
+  mustPayToScript
+    valHash
+    ( Datum $ toData $
+        MarketplaceDatum { getMarketplaceDatum: asset }
+    )
+    (Value.singleton (fst asset) (snd asset) one)
```

Comment:
How does balancing/min ADA requirement work? Will some balancing mechanism add missing ADA here?

Reply:
yes, it does, as the plutip tests pass, but I'll add it in explicit regardless